virtcontainers/filesystem.go

```diff
@@ -617,6 +655,35 @@ func (fs *filesystem) fetchAgentState(sandboxID string, state interface{}) error
 	return fs.fetchResource(true, sandboxID, "", agentFileType, state)
 }
 
+func (fs *filesystem) fetchContainersMap(ctrsMap *map[string][]string) error {
+	resource := containersMapFileType
+
+	filePath, _, err := fs.globalSandboxURI(resource)
+	if err != nil {
+		return err
+	}
+
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		f, err := os.Create(filePath)
+		if err != nil {
+			return err
+		}
+		f.Close()
+
+		m := make(map[string][]string)
+		if err := fs.storeContainersMap(m); err != nil {
+			return err
+		}
+	}
+
+	fileData, err := ioutil.ReadFile(filePath)
```

Comment:
This code implicitly only handles the current on-disk format of the containers map. We need to either:

- Add a statement in the code explaining that.

- Add something like a `version` field to the on-disk JSON format and have the code deal with any "older" versions it finds.